src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -49,10 +50,11 @@ public static void generateReposReport(List<RepoConfiguration> configs, String o
         for (RepoConfiguration config : configs) {
             Path repoReportDirectory = Paths.get(outputPath, config.getDisplayName());
             try {
-                GitDownloader.downloadRepo(config);
                 FileUtil.createDirectory(repoReportDirectory);
+                GitDownloader.downloadRepo(config);
             } catch (GitDownloaderException gde) {
                 logger.log(Level.WARNING, "Exception met while trying to clone the repo, will skip this one", gde);
+                generateEmptyRepoReport(repoReportDirectory.toString());
                 continue;
             } catch (IOException ioe) {
                 logger.log(Level.WARNING, "Error while creating repo directory, will skip this repo.", ioe);
```

Comment:
Should we also handle 'failing to create the directory' exception, by removing the config from the `configs` list?